src/Bicep.LangServer.IntegrationTests/BicepConfigTests.cs

```diff
@@ -0,0 +1,568 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Threading.Tasks;
+using Bicep.Core.FileSystem;
+using Bicep.Core.UnitTests.Assertions;
+using Bicep.Core.UnitTests.Utils;
+using Bicep.LangServer.IntegrationTests.Helpers;
+using Bicep.LanguageServer;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Extensions.LanguageServer.Protocol.Client;
+using OmniSharp.Extensions.LanguageServer.Protocol.Document;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using OmniSharp.Extensions.LanguageServer.Protocol.Workspace;
+using Range = OmniSharp.Extensions.LanguageServer.Protocol.Models.Range;
+
+namespace Bicep.LangServer.IntegrationTests
+{
+    [TestClass]
+    [SuppressMessage("Style", "VSTHRD200:Use \"Async\" suffix for async methods", Justification = "Test methods do not need to follow this convention.")]
+    public class BicepConfigTests
```

Comment:
A worthwhile test to add would be:
1. Create a mock `IFileResolver`
1. Make a bunch of changes to various file
1. Assert that the expected number of file reads for the config occurs

This would be a very good way to catch regressions - e.g. if we accidentally start reading from disk when we don't expect to, or read the same file multiple times unexpectedly.

Happy to help with the details if you want to discuss offline.

Reply:
As discussed offline, we'll only support reading from disk. I'll made changes to make sure we only read from disk when bicepconfig.json file is created, deleted and saved. I'll try to add a test to make sure we don't read config file on .bicep file changes

Reply:
Added tests to verify FileResolver.FileExists(..) is called only when required:
https://github.com/Azure/bicep/commit/5376c826f3082a42ace16a48f6427ca5c2c5ae6b

Let me know if this works.